# 1 The objective of this notebook is to train and test a baseline model - Logistic Regression

Reference metric: 73%

In [26]:
#---------Importing libraries---------#

#---Data analysis---#
import pandas as pd
import numpy as np


#---Data splitting---#
from sklearn.model_selection import train_test_split

#---classification models---#
from sklearn.linear_model import LogisticRegression

#---evaluation---#
from sklearn.metrics import accuracy_score, classification_report

#---utils---#
import os


## 1.1 Raw data

In [27]:
#importing raw data
df = pd.read_csv('data/ACME-HappinessSurvey2020.csv')

#splitting data into 80% train and 20% test
X = df.drop(columns=['Y'])
y = df['Y']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


In [28]:
LR = LogisticRegression(random_state=42)
LR.fit(X_train, y_train)

y_pred = LR.predict(X_test)

print('Base line Accuracy score: {:.2%} '.format(accuracy_score(y_test, y_pred)))

print(classification_report(y_test, y_pred))

Base line Accuracy score: 46.15% 
              precision    recall  f1-score   support

           0       0.56      0.33      0.42        15
           1       0.41      0.64      0.50        11

    accuracy                           0.46        26
   macro avg       0.48      0.48      0.46        26
weighted avg       0.49      0.46      0.45        26



Okay, our baseline is 46% accuracy. Let's see if we can do better.

## 1.2 Data version 1

In [29]:
#importing data
df_v1 = pd.read_csv('data/base_v1.csv')

#splitting data into 80% train and 20% test
X = df_v1.drop(columns=['Y'])
y = df_v1['Y']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [30]:
df_v1.shape

(109, 7)

In [31]:
LR = LogisticRegression(random_state=42)
LR.fit(X_train, y_train)

y_pred = LR.predict(X_test)

print('Base line Accuracy score: {:.2%} '.format(accuracy_score(y_test, y_pred)))

print(classification_report(y_test, y_pred))

Base line Accuracy score: 72.73% 
              precision    recall  f1-score   support

           0       0.83      0.50      0.62        10
           1       0.69      0.92      0.79        12

    accuracy                           0.73        22
   macro avg       0.76      0.71      0.71        22
weighted avg       0.75      0.73      0.71        22



72.7% ! almost a 30% improvement over the baseline. Let's see if we can do better with other models.